Adding null checks to dialogue conditions

SlyDevil opened the pull request:
CS + serve as soldier was causing crashes at the end of SaS events when opening dialogues.
Npc notables that don't have a map faction were sending nulls into the dialogue conditions.

There were also rare cases where npcs with factions like a kingdom leader were getting a null from .MapFaction, but those are covered by the same fix anyways.

SlyDevil committed:
added null check to IsHostile to try and address a SaS crash during random events

SlyDevil committed:
updated changelog for the fix

SlyDevil committed:
added additional null checks to references of MapFaction
updated changelog

SlyDevil committed:
Merge pull request #1 from SlyDevil/testing-sasfix

Testing sasfix

JoeFwd commented on changelog.txt:
```diff
@@ -1,3 +1,6 @@
+Version: 2.0.1
+Game Versions: v1.2.x
+* Fix: Added null checks for references to MapFaction in Dialogue Interpretor. Compatability with SaS
```
This changelog will be read by the users on release. So, we should reformulate the patch note so it is not too technical.

Something like `Fixed a crash when starting an interaction with NPCs without a faction. This improves compatibility with other mods such as Serve as a Soldier.` I'll let you adjust it.

JoeFwd commented:
Thanks for contributing to the project!

Kingdom leaders with no faction sounds like an issue to me. Do you know the root cause of this behaviour ? Destroyed kingdom ?

SlyDevil commented:
The underlying bug is rare as it is, in a couple hundred hours I ran into it maybe a half dozen times. All attempts to load back to a previous save and go into the same conversation would fail to reproduce the bug even if the save was right before; to the point that if people would report the crash in the ER discord I'd just tell them to reload a save a continue.

The one leader it happened to before I forked the project was Monchug and the khuzaits were in no way close to dying. Some of the other npcs that I ran into it with was SaS' weaponsmith, but those npcs are called from the culture file and I'd speak to them somewhat often, so running into a dialogue once a week that crashes with an npc who doesn't crash for hours left me at a dead end for reproduction.

SlyDevil committed:
Update changelog.txt

rephrased changelog note to be less technical; explicited the benefit of the fix

SlyDevil commented on changelog.txt:
```diff
@@ -1,3 +1,6 @@
+Version: 2.0.1
+Game Versions: v1.2.x
+* Fix : Factionless npcs dialogues should no longer cause crashes. Improves compatibility with non-Custom Spawns mods.
```
New formulation. Opted for "should" over "will" because I don't like the absolute formulation; I'm not *that* certain of the fix.
